saq/queue/postgres_asyncpg.py

```diff
@@ -0,0 +1,685 @@
+"""
+Postgres Queue using asyncpg
+"""
+
+from __future__ import annotations
+
+import asyncio
+import json
+import logging
+import time
+from collections.abc import AsyncGenerator
+from contextlib import asynccontextmanager
+from textwrap import dedent
+from typing import TYPE_CHECKING, Any, AsyncIterator, List, TypeVar
+
+from saq.errors import MissingDependencyError
+from saq.job import Job, Status
+from saq.multiplexer import Multiplexer
+from saq.queue.base import Queue, logger
+from saq.queue.postgres_ddl import DDL_STATEMENTS
+from saq.types import ListenCallback
+from saq.utils import now, seconds
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from saq.types import CountKind, DumpType, LoadType, QueueInfo, QueueStats
+try:
+    from asyncpg import Pool, create_pool, Connection
+    from asyncpg.exceptions import ConnectionDoesNotExistError, InterfaceError
+    from asyncpg.pool import PoolConnectionProxy
+
+except ModuleNotFoundError as e:
+    raise MissingDependencyError(
+        "Missing dependencies for Postgres. Install them with `pip install saq[asyncpg]`."
+    ) from e
+
+CHANNEL = "saq:{}"
+ENQUEUE = "saq:enqueue"
+DEQUEUE = "saq:dequeue"
+JOBS_TABLE = "saq_jobs"
+STATS_TABLE = "saq_stats"
+
+ContextT = TypeVar("ContextT")
+
+
+class PostgresAsyncpgQueue(Queue):
+    """
+    Queue is used to interact with Postgres using asyncpg.
+    """
+
+    @classmethod
+    def from_url(  # pyright: ignore[reportIncompatibleMethodOverride]
+        cls: type[PostgresAsyncpgQueue],
+        url: str,
+        min_size: int = 4,
```

Comment:
There's no obvious way to resize a queue in `asyncpg`.  Instead of having `min_size` and `max_size` as an init method, the are named arguments here.